### PR TITLE
Add ESS - Environmental Sensor System

### PR DESCRIPTION
**Pull request by dmills-lsst**
New CSC for rpi based sensor telemetry

### CODE REVIEW

**Comment by dmills-lsst** [comment]
Failing because subsystem count has increased

File: sal_interfaces/ESS/ESS_Telemetry.xml [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+          <Units>degreesC</Units>
+          <Conversion></Conversion>
+          <Sensor_location></Sensor_location>
+          <Count>1</Count>
+      </item>
+      <item>
+          <EFDB_Name>skyTemperature</EFDB_Name>
+	  <Description>Sky Temperature Sensor Telemetry</Description>
+          <IDL_Type>float</IDL_Type>
+          <Units>degrees</Units>
```
Comment (pkubanek) [inline sal_interfaces/ESS/ESS_Telemetry.xml]: should be degreesC

File: sal_interfaces/ESS/ESS_Telemetry.xml [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+          <Conversion></Conversion>
+          <Sensor_location></Sensor_location>
+          <Count>1</Count>
+      </item>
+      <item>
+          <EFDB_Name>temperatureC08</EFDB_Name>
+	  <Description>Temperature Sensor 8 Telemetry</Description>
+          <IDL_Type>float</IDL_Type>
+          <Units>degrees</Units>
```
Comment (pkubanek) [inline sal_interfaces/ESS/ESS_Telemetry.xml]: degreesC?

File: sal_interfaces/ESS/ESS_Telemetry.xml [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+          <Units>degreesC</Units>
+          <Conversion></Conversion>
+          <Sensor_location></Sensor_location>
+          <Count>1</Count>
+      </item>
+      <item>
+          <EFDB_Name>temperatureC02</EFDB_Name>
+	  <Description>Temperature Sensor 2 Telemetry</Description>
+          <IDL_Type>float</IDL_Type>
+          <Units>degrees</Units>
```
Comment (pkubanek) [inline sal_interfaces/ESS/ESS_Telemetry.xml]: Cosnistency! Either degreesC or degrees, but not both. My preference is for degreesC

**Comment by dmills-lsst** [comment]
This is sourced from the vendor XML. Rob , what is the approved temperature unit semantics that you check for?

**Comment by rbovill** [comment]
> This is sourced from the vendor XML. Rob , what is the approved temperature unit semantics that you check for?

I validate against Astropy valid units, so both degrees and degreesC are valid, iirc.  There might be a reason for the different temperatures, if the device is using different sensors?  Otherwise, we just want consistency for consistency's sake.

File: sal_interfaces/ESS/ESS_Telemetry.xml [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+ <Subsystem>ESS</Subsystem>
+ <Version>3.5.0</Version>
+ <Author>salgenerator</Author>
+ <EFDB_Topic>ESS_airTurbulence</EFDB_Topic>
+      <Description>Air Turbulence Instrument - CSAT3B (Tri-Axial) Serial to Ethernet Converter</Description>
+      <item>
+          <EFDB_Name>ux</EFDB_Name>
+	  <Description>x-axis wind speed</Description>
+          <IDL_Type>float</IDL_Type>
+          <Units>unitless</Units>
```
Comment (pkubanek) [inline sal_interfaces/ESS/ESS_Telemetry.xml:17]: I expect m/s or something similar..unitless isn't too helpfull

**Comment by dmills-lsst** [comment]
Used deg_C as units as the tests complained about degreeC

File: sal_interfaces/ESS/ESS_Telemetry.xml [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+          <Units>deg_C</Units>
+          <Conversion></Conversion>
+          <Sensor_location></Sensor_location>
+          <Count>1</Count>
+      </item>
+      <item>
+          <EFDB_Name>lightLevel</EFDB_Name>
+	  <Description>Light Level Telemetry</Description>
+          <IDL_Type>float</IDL_Type>
+          <Units>unitless</Units>
```
Comment (pkubanek) [inline sal_interfaces/ESS/ESS_Telemetry.xml:267]: candela?

File: sal_interfaces/ESS/ESS_Telemetry.xml [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+ <Subsystem>ESS</Subsystem>
+ <Version>3.5.0</Version>
+ <Author>salgenerator</Author>
+ <EFDB_Topic>ESS_airFlow</EFDB_Topic>
+      <Description>AirFlow Sensor - Windsonic1-L (Bi-axial) Serial to Ethernet Converter</Description>
+      <item>
+          <EFDB_Name>direction</EFDB_Name>
+	  <Description>Wind direction Telemetry</Description>
+          <IDL_Type>short</IDL_Type>
+          <Units>unitless</Units>
```
Comment (pkubanek) [inline sal_interfaces/ESS/ESS_Telemetry.xml:77]: deg? rad? I expect something like this unit..

Comment (dmills-lsst) [reply sal_interfaces/ESS/ESS_Telemetry.xml:77]: I will fix this after feedback from vendor, type short does not seem to fit anything obvious

**Pull request merged**